### PR TITLE
plugin: noise_suppression_interface: Query available devices

### PR DESCRIPTION
**Pull request by ranj063**
Query the list of available devices and use the NPU if available, otherwise compile the model for the CPU.

### CODE REVIEW

**Review by yongzhi1 — APPROVED** [review]
This is good, but I still prefer a way to override the device, just for testing convenience.

**Comment by ranj063** [comment]
> This is good, but I still prefer a way to override the device, just for testing convenience.

@yongzhi1 if you want to override the deivce name, lets wait for the kcontrol support in the plugin and we do it through that. Environment variables arent the way to go

File: tools/plugin/modules/ov_noise_suppression/noise_suppression_interface.cpp [hunk 1]
```diff
@@ -67,10 +68,19 @@ extern "C" {
...
 		/*
-		 * compile the model for the CPU and save the infer_request objects for each
-		 * channel separately
+		 * query the list of available devices and use NPU if available, otherwise use
+		 * CPU by default
 		 */
-		ov::CompiledModel compiled_model = core.compile_model(nd->model, "CPU", {});
+		std::vector<std::string> available_devices = core.get_available_devices();
+		for (auto &s: available_devices) {
+			if (!s.compare("NPU")) {
```
Comment (kv2019i) [inline tools/plugin/modules/ov_noise_suppression/noise_suppression_interface.cpp:76]: minor: For readbility, I'd prefer "if (s == "NPU")

**Review by kv2019i — APPROVED** [review]
Minor style nit, but no objections otherwise.

**Review by lgirdwood — APPROVED** [review]

**Pull request merged**